src/app/trainees/trainee-search/trainee-search.component.html

```diff
@@ -0,0 +1,14 @@
+<form (ngSubmit)="checkForm()" [formGroup]="form" autocomplete="off">
+  <mat-form-field>
+    <input
+      matInput
+      placeholder="Enter name / GMC no"
+      [formControlName]="'searchQuery'"
+      [value]="searchQuery$ | async"
+    />
+    <mat-error *ngIf="form.invalid">Minimum 3 characters needed</mat-error>
```

Comment:
unsure of min chars validation needed. Chinese doctor named QI or YI cannot be searched for? know the requirement said don't auto-search until 3 characters but this is just my thoughts

Reply:
will approve on your request. this is just food for thought. lemme know if its a go?

Reply:
good point, i'll reach out and ask Ify on the jira ticket

Reply:
Let me know when you're ready with this as well .... As for form control default auto search I'd suggest something along the lines of 
`searchQuery.valueChanges
      .pipe(debounceTime(1000))
      .subscribe((searchTerm: any) => {
        // do magic to put search term in url
      });`
I await till Ify's revision and your response

Reply:
yea i was thinking about using `valueChanges` on the form control as well but felt subscribing to every key stroke [even with a debounce] was not needed. at the moment if the mouse focus is on the form the enter key or submit button both run the `checkForm()` method which basically trims the search string and then if form is valid allows to proceed with dispatching the action and then hitting the BE.

ify updated the ticket earlier and changed the min character length to 2 - pr is updated to reflect